src/tracing/builder/parity.rs

```diff
@@ -224,53 +224,59 @@ impl ParityTraceBuilder {
 
         let vm_trace = trace_types.contains(&TraceType::VmTrace).then(|| self.vm_trace());
 
-        let traces = trace_types.contains(&TraceType::Trace).then(|| {
-            let mut traces = Vec::with_capacity(self.nodes.len());
-            // Boolean marker to track if sorting for selfdestruct is needed
-            let mut sorting_selfdestruct = false;
-
-            for node in self.iter_traceable_nodes() {
-                let trace_address = self.trace_address(node.idx);
-                let trace = node.parity_transaction_trace(trace_address);
-                traces.push(trace);
-
-                if node.is_selfdestruct() {
-                    // selfdestructs are not recorded as individual call traces but are derived from
-                    // the call trace and are added as additional `TransactionTrace` objects in the
-                    // trace array
-                    let addr = {
-                        let last = traces.last_mut().expect("exists");
-                        let mut addr = Vec::with_capacity(last.trace_address.len() + 1);
-                        addr.extend_from_slice(&last.trace_address);
-                        addr.push(last.subtraces);
-                        last.subtraces += 1;
-                        addr
-                    };
-
-                    if let Some(trace) = node.parity_selfdestruct_trace(addr) {
-                        traces.push(trace);
-                        sorting_selfdestruct = true;
-                    }
-                }
-            }
-
-            // Sort the traces only if a selfdestruct trace was encountered
-            if sorting_selfdestruct {
-                traces.sort_unstable_by(|a, b| a.trace_address.cmp(&b.trace_address));
-            }
-            traces
-        });
+        let traces = trace_types.contains(&TraceType::Trace).then(|| self.transaction_traces());
 
         let diff = with_diff.then(StateDiff::default);
 
         (traces, vm_trace, diff)
     }
 
+    /// Returns all the ordered [`TransactionTrace`], including selfdestructs.
+    ///
+    /// Selfdestructs appear as individual [`TransactionTrace`] instance but selfdestructs are
+    /// tracked as metadata of the recorded nodes.
+    fn transaction_traces(&self) -> Vec<TransactionTrace> {
```

Comment:
I only extracted this so I can properly document this.

and originally I thought we'd need to use this but we can also solve this with the iter, so no functional changes here